File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

double letter(string txt);
double sentence(string txt);
int main(void)

{
    string text =  get_string("Text: ");
    int index;

    // Coleman-Liau index formula
    index = round(0.0588 * letter(text) - 0.296 * sentence(text) - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

double letter(string txt)
{
    long L = 0;
    long w = 1;

    for (int i = 0; i < strlen(txt); i++)
    {
        // Get number of letters
        if ((txt[i] >= 'A' && txt[i] <= 'Z') || (txt[i] >= 'a' && txt[i] <= 'z'))
        {
            L++;
        }
        // Get number of words
        if (txt[i] == ' ')
        {
            w++;
        }
    }
    // Get the average letters in 100 words
    double avg = ((double) L / (double) w) * 100;
    return avg;
}

double sentence(string txt)
{
    long w = 1;
    long s = 0;

    for (int i = 0; i < strlen(txt); i++)
    {
        // Get number of words
        if (txt[i] == ' ')
        {
            w++;
        }
        // Get number of sentences
        if (txt[i] == '.' || txt[i] == '?' || txt[i] == '!')
        {
            s++;
        }
    }
    // Get the average sentences in 100 words
    double avg = ((double) s / (double) w) * 100;
    return avg;
}

File: sentimental-readability/readability.py
import cs50
import string
import re

text = cs50.get_string('Text: ')

# Obtaining letters
letters = list(re.sub("[?.,'! ]", '', text))
letter_count = len(letters)

# Obtaining words
words = text.split(' ')
word_count = len(words)

# Obtaining Sentences
sentences = re.split('[?.!]', text)
if sentences[-1] == '':
    sentences.pop(-1)
sentence_count = len(sentences)

# Average number of letter per 100 word
L = letter_count * (100 / word_count)

# Average number of sentences per 100 words
S = sentence_count * (100 / word_count)

# Coleman-Liau index
index = 0.0588 * L - 0.296 * S - 15.8

if index < 1:
    print('Before Grade 1')
elif index > 16:
    print('Grade 16+')
else:
    print(f'Grade {round(index)}')

